crates/hermesllm/src/apis/amazon_bedrock.rs

```diff
@@ -0,0 +1,1139 @@
+        }
+    }
+
+    fn from_endpoint(endpoint: &str) -> Option<Self> {
+        if endpoint.ends_with("/converse") {
+            Some(AmazonBedrockApi::Converse)
+        } else if endpoint.ends_with("/converse-stream") {
+            Some(AmazonBedrockApi::ConverseStream)
+        } else {
+            None
+        }
+    }
+
+    fn supports_streaming(&self) -> bool {
+        match self {
+            AmazonBedrockApi::Converse => false,
+            AmazonBedrockApi::ConverseStream => true,
+        }
+    }
+
+    fn supports_tools(&self) -> bool {
+        // Converse API has native tool support
+        true
+    }
+
+    fn supports_vision(&self) -> bool {
+        // Converse API has native vision support
+        true
+    }
+
+    fn all_variants() -> Vec<Self> {
+        vec![
+            AmazonBedrockApi::Converse,
+            AmazonBedrockApi::ConverseStream,
+        ]
+    }
+}
+
+// ============================================================================
+// CONVERSE REQUEST STRUCTURES
+// ============================================================================
+
+/// Amazon Bedrock Converse request
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ConverseRequest {
+    /// The model ID or ARN to invoke
+    pub model_id: String,
+    /// The messages to send to the model
+    pub messages: Option<Vec<Message>>,
+    /// System prompts that provide instructions or context
+    pub system: Option<Vec<SystemContentBlock>>,
+    /// Inference configuration
+    #[serde(rename = "inferenceConfig")]
+    pub inference_config: Option<InferenceConfiguration>,
+    /// Tool configuration for function calling
+    #[serde(rename = "toolConfig")]
+    pub tool_config: Option<ToolConfiguration>,
+    /// Guardrail configuration
+    #[serde(rename = "guardrailConfig")]
+    pub guardrail_config: Option<GuardrailConfiguration>,
+    /// Additional model-specific request fields
+    #[serde(rename = "additionalModelRequestFields")]
+    pub additional_model_request_fields: Option<Value>,
+    /// Additional model response field paths to return
+    #[serde(rename = "additionalModelResponseFieldPaths")]
+    pub additional_model_response_field_paths: Option<Vec<String>>,
+    /// Performance configuration
+    #[serde(rename = "performanceConfig")]
+    pub performance_config: Option<PerformanceConfiguration>,
+    /// Prompt variables for Prompt management
+    #[serde(rename = "promptVariables")]
+    pub prompt_variables: Option<HashMap<String, PromptVariableValues>>,
+    /// Request metadata for filtering logs
+    #[serde(rename = "requestMetadata")]
+    pub request_metadata: Option<HashMap<String, String>>,
+    /// Additional custom metadata (for internal use)
+    pub metadata: Option<HashMap<String, Value>>,
+    /// Whether this request should use streaming endpoint (internal field, not serialized)
+    #[serde(skip)]
+    pub stream: bool,
+}
```

Comment:
But stream=true/false message is part of ConverseRequest. When upstream is amazon_bedrock wouldn't you want to pass stream=true/false?